### ISSUE TITLE
[Bug?] Route for a Widget with default values. "Values can not be null"

### ISSUE THREAD

**Issue by MSchmack** [Milad-Akarie/auto_route_library#691]
A page with parameters that has default values with non nullable variables creates a route that complains that the value can not be null. 
The documentation states the following: 
`Note: Default values are respected. Required fields are also respected and handled properly.`

![Code_jWs1xcDkqn](https://user-images.githubusercontent.com/37783519/132827312-35eb27a6-ba0d-4aa8-a5b8-8ab28a06eed0.png)
Page Widget left, Route generated right

**Comment by RicardoRB** [comment]
Same here

**Comment by Milad-Akarie** [comment]
@MSchmack @Brazol @RicardoRB this analyzer related bug has been fixed in analyzer v 2.0.0, are you overriding the analyzer dependency in your project? please make sure you're using analyzer 2.0.0+.

**Comment by MSchmack** [comment]
@Milad-Akarie 
tl;dr solved with the higher version of analyzer (validated with 2.2.0).

I wasn't overriding the analyzer explicitly but a "old" dependency ("json_serializable@4.1.3") seems to have lowered the analyzer in the background to 1.7.x. After upgrading the dependency ("json_serializable@5") the flutter pub get command upgraded the analyzer in the pubspec lock file to 2.2.0

**Comment by Milad-Akarie** [comment]
@MSchmack  alright then. I'll try pin this for reference
